Enemies can attack breakable blocks

niamu opened the pull request:
@8bitgentleman has expressed a need for this a few times and will help him with #2347 and #2233.

This will likely require more testing, but this allows enemies to attack and deal damage to breakable blocks.

I tested this by adding one of the breakable blocks from the forest into the hallway level. Hippies attack the block until it is broken and then proceed as normal. The process should be similar for all other enemies.

8bitgentleman commented:
Thanks for this @niamu !  Couple things I noticed:
1. When the block is destroyed there are some strange sounds, especially if the block has death sound effect
2. Coming from the right the hippy sprites flip when they touch the block
3. When an enemy attack there is a small period of time when the player is invincible so that the enemy doesn't do constant damage.  This doesn't happen when enemies attack these blocks.  I'm not sure if it should happen but it's worth talking about.
4. Maybe there should be a way to make blocks invulnerable to enemies?  Not sure where exactly this would be used but it might be useful in the future.
5. Enemies breaking blocks can also lead to interesting things.  In Valley-3 it's possible to lure the guitarist onto the breakable blocks where he falls to his death.
6. I checked all levels with breakable blocks and they all function as they should ( other than the interesting way to kill the guitarist in valley-3.

niamu commented:
I believe I've addressed all of these concerns now with the exception of the guitarist in valley-3.

If I write some code that doesn't let the guitarist attack those blocks while on top of them, then it will likely break your plans for having bosses smash through walls in the future.

It should be fixed somehow, I'm just not sure what the best solution is.

8bitgentleman commented:
When attacking the block the hippie repeats an audio file over and over, also the hippies on the left now flip when attacking the block.

niamu commented:
Resolved the problem of enemies repeating their attack sound. As for the flipping problem, it's actually a separate bug related to how enemies interact with walls and not something that I think I can fix in this pull request.

8bitgentleman commented:
LGTM

CalebJohn commented:
LGTM
